### DISCUSSION TITLE
Welcome to Flask-Muck Discussions!

### DISCUSSION THREAD

**Discussion by dtiesling** [dtiesling/flask-muck#12] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other users. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other users.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by BrianBut** [comment]
~/Sandbox/.venv/lib/python3.11/site-packages/marshmallow_jsonschema/__init__.py:1: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.

I am confused by this message which appears whenever I run flask. Is it important? How to bypass it?

**Reply by dtiesling** [reply]
Safe to ignore and I'll look at updating those packages in a future release. I already need to push out a release for SQLA 1.4 support so I'll do that all together.

**Reply by BrianBut** [reply]
I have since discovered that this issue prevents the use of the Flask unittest package as the warning  causes unittests to fail. 
I think the issue should be treated as a bug.

**Reply by dtiesling** [reply]
Yep, it's broken the build process as well. 

Unfortunately I found out the dependency on pkg_resources is coming from a different library. I've submitted a PR to the maintainer of that repository. If they can merge that change and push up a new release to pypi the tests and builds will be working again. If not I'll need to fork that code and push up my own version. Should 
be fixed this week some time either way.

**Reply by BrianBut** [reply]
I have been able to circumvent the issue by installing from this requirements.txt file.
[requirements.txt](https://github.com/user-attachments/files/21502700/requirements.txt)

**Reply by dtiesling** [reply]
This should be resolved in version 0.4.2

**Reply by BrianBut** [reply]
I still have not been able to make it work. 

I tested on your dtiesling/introducing-flask-muck-how-to-build-a-comprehensive-flask-rest-api-in-5-minutes.
I still get the same error as I got yesterday.
  File "/home/brian/SandboxPydantic/sample_api/.venv/lib/python3.11/site-packages/flask_muck/views.py", line 181, in FlaskMuckApiView
    "limit": fields.Integer(missing=None),
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/brian/SandboxPydantic/sample_api/.venv/lib/python3.11/site-packages/marshmallow/fields.py", line 977, in __init__
    super().__init__(as_string=as_string, **kwargs)
  File "/home/brian/SandboxPydantic/sample_api/.venv/lib/python3.11/site-packages/marshmallow/fields.py", line 924, in __init__
    super().__init__(**kwargs)
TypeError: Field.__init__() got an unexpected keyword argument 'missing'

It looks to me as if line 181 of flask_muck/views.py is calling a 'missing' parameter which marshmallow/fields.py does not recognise.

**Reply by dtiesling** [reply]
I'll go back and have a look at that article. It may be out of date, it's from a couple years ago. In the meantime you can check out the examples in the repo. Those should be working.

**Reply by BrianBut** [reply]
It works perfectly if run from the repo when run from the repos envelope.
The difficulties arise when any of the examples are copied elsewhere. 
There is a Pipfile and Pipfile.lock in each of the examples, but I have not been able to use pipenv install.
I can make it work using pip install, stipulating marshmallow==3.26.1 and letting pip solve the resulting conflicting requirements.
I am not yet comfortable with poetry.

**Reply by dtiesling** [reply]
Looks like they removed the "missing" kwarg in marshmallow 4.0. You can remove that kwarg or replace it with load_default and dump_default

**Reply by BrianBut** [reply]
I have made the required changes.

I have a modified 'flask-muck' in git@github.com:BrianBut/my-flask-muck.git
This removes the warnings except  for two. One of the remaining is in the marshmallow_jsonschema package, and the other is a query join issue which do not yet understand.

I would like to push it to flask-muck, but I don't know where to go from here. Help!
